integration_tests/test_search.py

```diff
@@ -209,19 +232,46 @@ def _match_target_age(self, how_old_is_the_kid, result):
             return True, how_old_is_the_kid, (result_lower, result_upper)
         return False, how_old_is_the_kid, (result_lower, result_upper)
 
+    def _match_author(self, author, result):
```

Comment:
This encapsulates the logic for checking that a particular author was a major contributor to a book.